sourcegen-generator-java/src/main/java/io/micronaut/sourcegen/JavaPoetSourceGenerator.java

```diff
@@ -538,8 +538,9 @@ private CodeBlock renderExpression(@Nullable ObjectDef objectDef, MethodDef meth
         }
         if (expressionDef instanceof ExpressionDef.Cast castExpressionDef) {
             return CodeBlock.concat(
-                CodeBlock.of("($T) ", asType(castExpressionDef.type(), objectDef)),
-                renderExpression(objectDef, methodDef, castExpressionDef.expressionDef())
+                CodeBlock.of("($T) (", asType(castExpressionDef.type(), objectDef)),
+                renderExpression(objectDef, methodDef, castExpressionDef.expressionDef()),
+                CodeBlock.of(")")
```

Comment:
As Elif mentioned to me, it is hard to change order of operations in expressions and maybe a way of adding brackets would be good.